homeassistant/components/insteon/config_flow.py

```diff
@@ -0,0 +1,353 @@
+
+
+class InsteonOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle an Insteon options flow."""
+
+    def __init__(self, config_entry):
+        """Init the InsteonOptionsFlowHandler class."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Init the config flow."""
+        errors = {}
+        if user_input is not None:
+            change_hub_config = user_input.get(STEP_CHANGE_HUB_CONFIG, False)
+            device_override = user_input.get(STEP_ADD_OVERRIDE, False)
+            x10_device = user_input.get(STEP_ADD_X10, False)
+            remove_override = user_input.get(STEP_REMOVE_OVERRIDE, False)
+            remove_x10 = user_input.get(STEP_REMOVE_X10, False)
+            if _only_one_selected(
+                change_hub_config,
+                device_override,
+                x10_device,
+                remove_override,
+                remove_x10,
+            ):
+                if change_hub_config:
+                    return await self.async_step_change_hub_config()
+                if device_override:
+                    return await self.async_step_add_override()
+                if x10_device:
+                    return await self.async_step_add_x10()
+                if remove_override:
+                    return await self.async_step_remove_override()
+                if remove_x10:
+                    return await self.async_step_remove_x10()
+            errors["base"] = "select_single"
+
+        data_schema = {
+            vol.Optional(STEP_ADD_OVERRIDE): bool,
+            vol.Optional(STEP_ADD_X10): bool,
+        }
+        if self.config_entry.data.get(CONF_HOST):
+            data_schema[vol.Optional(STEP_CHANGE_HUB_CONFIG)] = bool
+
+        options = {**self.config_entry.options}
+        if options.get(CONF_OVERRIDE):
+            data_schema[vol.Optional(STEP_REMOVE_OVERRIDE)] = bool
+        if options.get(CONF_X10):
+            data_schema[vol.Optional(STEP_REMOVE_X10)] = bool
+
+        return self.async_show_form(
+            step_id="init", data_schema=vol.Schema(data_schema), errors=errors
+        )
+
+    async def async_step_change_hub_config(self, user_input=None):
+        """Change the Hub password."""
+        if user_input is not None:
+            data = {
+                **self.config_entry.data,
+                CONF_HOST: user_input[CONF_HOST],
+                CONF_PORT: user_input[CONF_PORT],
+            }
+            if {**self.config_entry.data}[CONF_HUB_VERSION] == 2:
+                data[CONF_USERNAME] = user_input[CONF_USERNAME]
+                data[CONF_PASSWORD] = user_input[CONF_PASSWORD]
+            self.hass.config_entries.async_update_entry(self.config_entry, data=data)
+            return self.async_create_entry(
+                title="", data={**self.config_entry.options},
+            )
+        curr_host = {**self.config_entry.data}[CONF_HOST]
```

Comment:
```suggestion
        curr_host = self.config_entry.data[CONF_HOST]
```

Do these need to be copies?